test/public_plans_interceptor_test/public_plans_test.go

```diff
@@ -488,4 +489,96 @@ var _ = Describe("Service Manager Public Plans Interceptor", func() {
 			})
 		})
 	})
+
```

Comment:
What about tests where there's a mix - a plan has supported platform types and another has supported platform names? This is a realistic scenario that will happen.

Reply:
Hi,

These tests are implemented in sm-sap.
Kindly see(within: notifications_test.go):

Context("Define a catalog with mixed metadata definitions: one plan with 'supportedPlatforms' and another with 'supportedPlatformNames'"

Context("Define a catalog illegally having metadata of both supportedPlatforms and supportedPlatformNames within a single plan"